### PR TITLE
Support import git repository

### PR DESCRIPTION
**Pull request by xuzhang3**
## All Submissions:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.
* [x] All new and existing tests passed.
* [x] My code follows the code style of this project.
* [x] I ran lint checks locally prior to submission.
* [x] Have you checked to ensure there aren't other open PRs for the same update/change?

## What about the current behavior has changed?

Issue Number: #43  #45 

## Does this introduce a change to `go.mod`, `go.sum` or `vendor/`?

- [ ] Yes
- [x] No

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Any relevant logs, error output, etc?

(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

### CODE REVIEW

File: website/docs/r/azure_git_repository.html.markdown [hunk 1]
```diff
@@ -55,10 +69,8 @@ The following arguments are supported:
 `initialization` - (Required) block supports the following:
 
 - `init_type` - (Required) The type of repository to create. Valid values: `Uninitialized` or `Clean`. Defaults to `Uninitialized`.
```
Comment (EliiseS) [inline website/docs/r/azure_git_repository.html.markdown]: Missing import mention

```suggestion
- `init_type` - (Required) The type of repository to create. Valid values: `Uninitialized`, `Clean` or `Import`. Defaults to `Uninitialized`.
```

File: azuredevops/internal/service/git/resource_git_repository.go [hunk 2]
```diff
@@ -160,6 +164,23 @@ func resourceGitRepositoryCreate(d *schema.ResourceData, m interface{}) error {
 		return fmt.Errorf("Error creating repository in Azure DevOps: %+v", err)
 	}
 
+	if strings.EqualFold(initialization.initType, string(RepoInitTypeValues.Import)) {
+		if strings.EqualFold(initialization.sourceType, "Git") {
```
Comment (EliiseS) [inline azuredevops/internal/service/git/resource_git_repository.go]: These two if statements can be merged

File: azuredevops/internal/service/git/resource_git_repository.go [hunk 3]
```diff
@@ -112,14 +112,18 @@ func ResourceGitRepository() *schema.Resource {
...
 						},
 						"source_type": {
-							Type:     schema.TypeString,
-							Optional: true,
-							Default:  "",
+							Type:         schema.TypeString,
+							Optional:     true,
+							ForceNew:     true,
+							ValidateFunc: validation.StringInSlice([]string{"Git"}, false),
+							Default:      "Git",
```
Comment (EliiseS) [inline azuredevops/internal/service/git/resource_git_repository.go:120]: Add required
```suggestion
							Default:      "Git",
							RequiredWith: []string{"initialization .0.source_url"},
```

File: azuredevops/internal/service/git/resource_git_repository.go [hunk 4]
```diff
@@ -112,14 +112,18 @@ func ResourceGitRepository() *schema.Resource {
...
 						},
 						"source_url": {
-							Type:     schema.TypeString,
-							Optional: true,
-							Default:  "",
+							Type:         schema.TypeString,
+							Optional:     true,
+							ForceNew:     true,
+							Default:      "",
+							ValidateFunc: validation.IsURLWithHTTPorHTTPS,
```
Comment (EliiseS) [inline azuredevops/internal/service/git/resource_git_repository.go:128]: Add required
```suggestion
							ValidateFunc: validation.IsURLWithHTTPorHTTPS,
							RequiredWith: []string{"initialization .0.source_typel"},
```

**Review by EliiseS — CHANGES_REQUESTED** [review]
Looks great! I got just a few change requests `:D

File: azuredevops/internal/service/git/resource_git_repository.go (hunk 2 above)
Comment (xuzhang3) [reply azuredevops/internal/service/git/resource_git_repository.go]: Agree with you. The reason why have two if statement is web UI support two types import: `git`, `tfvc`, I only implement the `git` import feature. :D

Comment (sdebruyn) [reply azuredevops/internal/service/git/resource_git_repository.go]: You forgot to check if `initialization` is nil here @xuzhang3

**Pull request merged**